src/engine/controls/cuecontrol.cpp

```diff
@@ -1173,6 +1208,44 @@ void CueControl::hotcueActivatePreview(HotcueControl* pControl, double value) {
     setHotcueFocusIndex(pControl->getHotcueIndex());
 }
 
+void CueControl::loopCueReloop(HotcueControl* pControl, const CuePointer& pCue) {
+    if (pCue->getType() != mixxx::CueType::Loop) {
+        return;
+    }
+
+    if (m_pCurrentSavedLoopControl != pControl) {
+        setCurrentSavedLoopControlAndActivate(pControl);
+    } else {
+        bool loopActive = pControl->getStatus() ==
+                HotcueControl::Status::Active;
+        Cue::StartAndEndPositions pos = pCue->getStartAndEndPosition();
+        setLoop(pos.startPosition, pos.endPosition, !loopActive);
+    }
+}
+
+void CueControl::loopCueGoToAndLoopOrToggle(HotcueControl* pControl, const CuePointer& pCue) {
+    if (pCue->getType() != mixxx::CueType::Loop) {
+        return;
+    }
+
+    if (m_pCurrentSavedLoopControl != pControl) {
+        setCurrentSavedLoopControl(pControl);
+    }
```

Comment:
Is here a return missing?

Reply:
No, this just sets the current active loop.
Below are the cases for jumping or toggling.